xapian-core/include/xapian/weight.h

```diff
@@ -526,15 +526,20 @@ class XAPIAN_VISIBILITY_DEFAULT BM25Weight : public Weight {
      *		   document length normalisation of wdf is.  0 means no
      *		   normalisation; 1 means full normalisation.  (default 0.5)
      *
+     *  @param delta  A parameter for pseudo tf value to control the scale
+     * 		   of the tf lower bound. Delta(δ) can be tuned for e.g from 0.0 
+     * 		   to 1.5 but BM25+ can still work effectively across collections 
+     *		   with a fixed δ = 1.0. (default 1.0)
+     * 
      *  @param min_normlen  A parameter specifying a minimum value for
      *		   normalised document length.  Normalised document length
      *		   values less than this will be clamped to this value, helping
      *		   to prevent very short documents getting large weights.
      *		   (default 0.5)
      */
-    BM25Weight(double k1, double k2, double k3, double b, double min_normlen)
+    BM25Weight(double k1, double k2, double k3, double b, double delta, double min_normlen)
```

Comment:
C++11 is OK inside the library, but we're trying to avoid it in the API headers for 1.4.x.  This is just for practical reasons, as at least the older versions of the compilers we aim to support have C++11 off by default - if our API headers use it, then users are forced to pass the appropriate option to their compiler, which there's no portable way to do, so you end up having to hard code for one compiler, or probe in `configure` or equivalent.

So far there's not been anything from C++11 that's really needed in the API headers - for this case, I'd just add a second implementation of the constructor for now (the obvious alternative is adding a new `delta` parameter and giving it a default value, but that would be an ABI-incompatible change so we couldn't include that in 1.4.x - user code built against 1.4.0 would fail to find the implementation of the 4-parameter form in a shared library built from 1.4.x with this change).  But beyond 1.4.x, adding a new `delta` parameter with default value 1.0 to the end of the parameter list should achieve this without needing C++11 enabled when building code using the library).